src/app/campaign.service.ts

```diff
@@ -72,10 +72,9 @@ export class CampaignService {
       return undefined;
```

Comment:
slightly offtopic since this is an unchanged line, but I guess we could return `NaN` here and then the return type of the function would be `number`. Maybe it doesn't help. Or make having a strictly positive target a precondition of calling this function and throw an error.